### PR TITLE
Include min/max/average file size in Searchable Snapshots Stats API

### PR DESCRIPTION
**Pull request by tlrx**
This pull request adds the minimum, maximum and average length of files in the Searchable Snapshots Stats API:

```
 {
      "file_ext" : "fdt",
      "num_files" : 31,
      "total_size" : 417698246269,
      "min_size" : "480mb",
      "min_size_in_bytes" : 503370620,
      "max_size" : "12.9gb",
      "max_size_in_bytes" : 13947816041,
      "avg_size" : "12.5gb",
      "avg_size_in_bytes" : 13474136976,
      ...
}
```

These information are useful in understanding why some data is still being read from the blob store when the cached blob in `.snapshot-blob-cache` should have been sufficient to server all the read operations.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/IndexInputStats.java [hunk 1]
```diff
@@ -137,14 +149,22 @@ public Releasable addIndexCacheFill() {
...
-    public int getNumFiles() {
+    public long getNumFiles() {
         return numFiles;
     }
 
     public long getTotalSize() {
         return totalSize;
     }
 
+    public ByteSizeValue getMinSize() {
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/IndexInputStats.java:160]: I would keep this a long in this class, or change totalSize also to use ByteSizeValue

**Review by ywelsch — COMMENTED** [review]
Very useful to add this info to the stats API. I've left some minor comments w.r.t. consistency between total and min/max fields. I'm also wondering we should fold `CacheIndexInputStats.{total_size,min_size,max_size,avg_size}` into a size section (i.e. have fields "total,min,max,avg" under a "size" object)

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/SearchableSnapshotShardStats.java [hunk 2]
```diff
@@ -334,6 +366,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 builder.field("file_ext", getFileExt());
                 builder.field("num_files", getNumFiles());
                 builder.field("total_size", getTotalSize());
```
[the remark this answers is not in the document]
Comment (tlrx) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/SearchableSnapshotShardStats.java:368]: I've changed the output to:
```
{
      "file_ext" : "dvd",
      "num_files" : 5,
      "open_count" : 5,
      "close_count" : 5,
      "size" : {
        "total" : "82gb",
        "total_in_bytes" : 88106037691,
        "min" : "16.3gb",
        "min_in_bytes" : 17607618817,
        "max" : "16.4gb",
        "max_in_bytes" : 17632870511,
        "average" : "16.4gb",
        "average_in_bytes" : 17621207538
      },
      ...
```

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/IndexInputStats.java (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/IndexInputStats.java]: I keep total/min/max as long in this class and they are later converted as ByteSizeValue when aggregated in the stats as CacheIndexInputStats.

**Comment by tlrx** [comment]
@ywelsch thanks for your feedback. I've updated the code, can you please have another look when you have time?

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by tlrx** [comment]
Thanks Yannick

**Pull request merged**